simplex/block.go

```diff
@@ -58,9 +66,53 @@ func (b *Block) Bytes() ([]byte, error) {
 	return cBlock.MarshalCanoto(), nil
 }
 
+// Verify verifies the block.
 func (b *Block) Verify(ctx context.Context) (simplex.VerifiedBlock, error) {
-	// TODO: track blocks that have been verified to ensure they are either rejected or accepted
-	return b, b.vmBlock.Verify(ctx)
+	if b.metadata.Seq != 0 {
+		err := b.verifyParentMatchesPrevBlock(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	err := b.vmBlock.Verify(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify block: %w", err)
+	}
+
+	// once verified, the block tracker will eventually either accept or reject the block
+	b.blockTracker.trackBlock(b)
+
+	return b, nil
+}
+
+// verifyPrevBlock verifies that the previous block referenced in the current block's metadata
+// matches the parent of the current block's vmBlock.
+func (b *Block) verifyParentMatchesPrevBlock(ctx context.Context) error {
+	prevBlock := b.blockTracker.getBlock(b.metadata.Prev)
+	if prevBlock != nil {
+		if b.vmBlock.Parent() != prevBlock.vmBlock.ID() {
+			return fmt.Errorf("%w: parentID %s, prevID %s", errMismatchedPrevDigest, b.vmBlock.Parent(), prevBlock.vmBlock.ID())
+		}
+
+		return nil
+	}
+
+	// if we do not have it in the map, it's possible for it to be the last accepted block
+	lastID, err := b.blockTracker.vm.LastAccepted(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get last accepted block: %w", err)
+	}
+
+	if lastID != b.vmBlock.Parent() {
```

Comment:
I would very much like us to explicitly make sure that the chain induced by Simplex is aligned with the chain induced by the VM, and not only the inner VMs are aligned:

```
prevBlock.BlockHeader().Digest == b.metadata.Prev && prevBlock.vmBlock.ID() == b.vmBlock.Parent()
```